cyberbrain/value_stack.py

```diff
@@ -201,35 +306,39 @@ def _tos(self, n):
     def _push(self, *values):
         """Pushes values onto the simulated value stack.
 
-        This method will automatically convert single value to a list. _placeholder will
-        be converted to an empty list, so that it never exists on the value stack.
-
-        Str is converted to Symbol.
+        StackItems are copied onto the stack
+        Str (name of variable) is converted to Symbol and then wrapped into a SymbolStackItem.
+        If it is none of the above, the value is stored as a custom value.
         """
+        start_lineno = self.last_starts_line
         for value in values:
-            if value is _placeholder:
-                value = []
+            sources = []
+            if isinstance(value, SymbolStackItem) or isinstance(
+                value, CustomValueStackItem
+            ):
+                newValue = value.copy()
+                self.stack.append(newValue)
```

Comment:
newValue -> new_value

Could you elaborate a bit on the reason to copy (instead of just pushing `value` onto the stack)?

Reply:
Same reason why we copy symbols, since later mutations could cause changes to previous symbols